lang/cs/Org.Apache.REEF.IMRU/API/IMRUJobDefinition.cs

```diff
@@ -27,23 +27,139 @@ namespace Org.Apache.REEF.IMRU.API
     /// <seealso cref="IMRUJobDefinitionBuilder" />
     public sealed class IMRUJobDefinition
     {
-        private readonly IConfiguration _configuration;
         private readonly string _jobName;
+        private readonly IConfiguration _mapFunctionConfiguration;
+        private readonly IConfiguration _mapInputCodecConfiguration;
+        private readonly IConfiguration _updateFunctionCodecsConfiguration;
+        private readonly IConfiguration _reduceFunctionConfiguration;
+        private readonly IConfiguration _updateFunctionConfiguration;
+        private readonly IConfiguration _mapOutputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _mapInputPipelineDataConverterConfiguration;
+        private readonly IConfiguration _partitionedDatasetConfiguration;
+        private readonly int _numberOfMappers;
 
-        internal IMRUJobDefinition(IConfiguration configuration, string jobName)
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="mapFunctionConfiguration">Map function configuration</param>
+        /// <param name="mapInputCodecConfiguration">Map input codec configuration</param>
+        /// <param name="updateFunctionCodecsConfiguration">codec configuration for update 
+        /// function. It is union of TMapInput, TMapOutput and TResult configuration</param>
+        /// <param name="reduceFunctionConfiguration">Reduce function configuration</param>
+        /// <param name="updateFunctionConfiguration">Update function configuration</param>
+        /// <param name="mapOutputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapOutput</param>
+        /// <param name="mapInputPipelineDataConverterConfiguration">Configuration of 
+        /// PipelineDataConverter for TMapInput</param>
+        /// <param name="partitionedDatasetConfiguration">Configuration of partitioned 
+        /// dataset</param>
+        /// <param name="numberOfMappers">Number of mappers</param>
+        /// <param name="jobName">Job name</param>
+        internal IMRUJobDefinition(
```

Comment:
To be more specific, say the two named parameters correspond to binding of configuration of IStreaming<TMapInput> and IStreaming<TMapOutput>. but they will also need to append this with base configurations config1 and config2 below.

 var config1 = TangFactory.GetTang().NewConfigurationBuilder()
                .BindImplementation(GenericType<IStreamingCodec<float>>.Class, GenericType<FloatStreamingCodec>.Class)
                .Build();
            var config2 = TangFactory.GetTang().NewConfigurationBuilder()
                .BindImplementation(GenericType<IStreamingCodec<float>>.Class, GenericType<FloatStreamingCodec>.Class)
                .Build();

So when now we merge together there is going to be conflict.

Are you saying that we put the whole configuration of TMapInput codec including the base one as one named parameter and the one corresponding to TMapOutput as another one.